includes/classes/Authentication/Passwords.php

```diff
@@ -374,4 +384,44 @@ public function enforce_for_user( $user_id ) {
 
 		return $enforce;
 	}
+
+	/**
+	 * Check if password is secure by querying the Have I Been Pwned API.
+	 *
+	 * @param string $password Password to validate.
+	 *
+	 * @return bool True if password is ok, false if it shows up in a breach.
+	 */
+	protected function is_password_secure( $password ): bool {
```

Comment:
**thought (non-blocking):** What are your thoughts on allowing engineers to opt-out of this functionality via a const or a filter?